drake/systems/plants/RigidBodyManipulator.h

```diff
@@ -97,46 +99,256 @@ class DLLEXPORT_RBM RigidBodyManipulator
   std::string getVelocityName(int velocity_num) const;
   std::string getStateName(int state_num) const;
 
-  template <typename DerivedQ, typename DerivedV>
-  void doKinematics(const MatrixBase<DerivedQ> &q, const MatrixBase<DerivedV> &v, bool compute_gradients = false, bool compute_JdotV = true);
+  template <typename DerivedQ>
+  KinematicsCache<typename DerivedQ::Scalar> doKinematics(const MatrixBase<DerivedQ>& q, int gradient_order) {
+    KinematicsCache<typename DerivedQ::Scalar> ret(bodies, gradient_order);
+    ret.initialize(q);
+    doKinematics(ret);
+    return ret;
+  }
 
-  bool isBodyPartOfRobot(const RigidBody& body, const std::set<int>& robotnum);
+  template <typename DerivedQ, typename DerivedV>
+  KinematicsCache<typename DerivedQ::Scalar> doKinematics(const MatrixBase<DerivedQ>& q, const MatrixBase<DerivedV>& v, int gradient_order = 0, bool compute_JdotV = true) {
+    KinematicsCache<typename DerivedQ::Scalar> ret(bodies, gradient_order);
+    ret.initialize(q, v);
+    doKinematics(ret, compute_JdotV);
+    return ret;
+  }
 
-  double getMass(const std::set<int>& robotnum = RigidBody::defaultRobotNumSet);
+  template <typename Scalar>
+  void doKinematics(KinematicsCache<Scalar>& cache, bool compute_JdotV = false) const {
```

Comment:
I was referring to the fact that compilers are also free to use inline expansion for any function that's not marked inline.
Defining a method in the class body itself (in the header) does the same thing as adding the inline keyword, yes, but as I said that's not the same thing as forcing the compiler to perform inline expansion.